cpp/src/io/parquet/writer_impl.cu

```diff
@@ -244,15 +258,17 @@ struct leaf_schema_fn {
   template <typename T>
   std::enable_if_t<std::is_same_v<T, int32_t>, void> operator()()
   {
-    col_schema.type        = Type::INT32;
-    col_schema.stats_dtype = statistics_dtype::dtype_int32;
+    col_schema.type           = Type::INT32;
+    col_schema.converted_type = ConvertedType::INT_32;
+    col_schema.stats_dtype    = statistics_dtype::dtype_int32;
   }
 
   template <typename T>
   std::enable_if_t<std::is_same_v<T, int64_t>, void> operator()()
   {
-    col_schema.type        = Type::INT64;
-    col_schema.stats_dtype = statistics_dtype::dtype_int64;
+    col_schema.type           = Type::INT64;
+    col_schema.converted_type = ConvertedType::INT_64;
```

Comment:
If these aren't being converted, why do we need to store a converted type for them now? Just to be explicit so we don't have to check later?

Reply:
I use the physical type and converted type to figure out which field of the stats union to use when figuring out the sort order of the data.  Converted type wasn't being set for the signed ints, and also wasn't being passed in the parquet_column_device_view. stats_dtype doesn't distinguish between signed and unsigned sadly.